pkg/services/ngalert/migration/channel.go

```diff
@@ -26,68 +24,42 @@ const (
 )
 
 // migrateChannels creates Alertmanager configs with migrated receivers and routes.
-func (om *OrgMigration) migrateChannels(channels []*legacymodels.AlertNotification) (*migmodels.Alertmanager, error) {
-	amConfig := migmodels.NewAlertmanager()
-	empty := true
+func (om *OrgMigration) migrateChannels(channels []*legacymodels.AlertNotification) ([]*migmodels.ContactPair, error) {
 	// Create all newly migrated receivers from legacy notification channels.
+	pairs := make([]*migmodels.ContactPair, 0, len(channels))
 	for _, c := range channels {
 		receiver, err := om.createReceiver(c)
 		if err != nil {
+			om.log.Warn("Failed to create receiver", "type", c.Type, "name", c.Name, "uid", c.UID, "error", err)
 			if errors.Is(err, ErrDiscontinued) {
-				om.log.Error("Alert migration error: discontinued notification channel found", "type", c.Type, "name", c.Name, "uid", c.UID)
+				// Don't fail on discontinued channels.
 				continue
 			}
 			return nil, fmt.Errorf("channel '%s': %w", c.Name, err)
 		}
 
-		empty = false
 		route, err := createRoute(c, receiver.Name)
 		if err != nil {
+			om.log.Warn("Failed to create route", "type", c.Type, "name", c.Name, "uid", c.UID, "error", err)
```

Comment:
Do we need warn here? It returns error and will (should) be logged upstream.

Reply:
True, this is mostly to get logs for discontinued channels which do not cause an error in the current version, I'll put it in the if statement and remove the log on the route creation.